main.go

```diff
@@ -73,11 +74,36 @@ func main() {
 		sigint := make(chan os.Signal, 1)
 		signal.Notify(sigint, os.Interrupt, syscall.SIGTERM)
 		<-sigint
-		s.stop <- struct{}{} // notify having caught signal
+		s.stop <- struct{}{}             // notify having caught signal stop oauthproxy
+		metricsStopChannel <- struct{}{} // and the metrics endpoint
 	}()
 	s.ListenAndServe()
 }
 
+// startMetricsServer will start the metrics server on the specified address.
+// It always return a channel to signal stop even when it does not run.
+func startMetricsServer(address string) chan struct{} {
+	stop := make(chan struct{}, 1)
+
+	// Attempt to setup the metrics endpoint if we have an address
+	if address != "" {
+		s := &http.Server{Addr: address, Handler: middleware.DefaultMetricsHandler}
+		go s.ListenAndServe()
```

Comment:
What happens if there is an error starting this up, eg and invalidly formatted address?

Reply:
Good point. I have wrapped this in `logger.Fatal()` now but I'm not sure if that is the right call. If the metrics address is invalid should it fail to start or just fail to run the metrics and display an error?